mbof/views.py

```diff
@@ -27,60 +28,107 @@ class CurrentUserViewSet(UserViewSet):
+    API endpoint that allows Events to be viewed or edited.
     """
     queryset = Event.objects.all().order_by('postingTime')
     serializer_class = EventSerializer
 
+    def perform_create(self, serializer):
+        me = self.request.user.__str__()
+        method = self.request.META['REQUEST_METHOD']
+        url = self.request.META['PATH_INFO']
+        # EXAMPLE - dovek : POST /api/events/
+        message = '%s : %s %s' % (me, method, url)
+        logger.info(message)
+        serializer.save(owner=me)
+
 
-class EventListViewSet(generics.ListAPIView):
+class EventListViewSet(generics.ListCreateAPIView):
     """
     API endpoint that allows users to be viewed or edited.
     """
     serializer_class = EventSerializer
 
     def get_queryset(self):
-        logger.debug('Type=' + self.kwargs['type'])
+        me = self.log_action()
+
+        query_set = self.queryset
         current_time = timezone.now()
         if self.kwargs['type'] is None:
-            return Event.objects.all().order_by('postingTime')
+            logger.debug('Type is None')
+            query_set = Event.objects.all().order_by('postingTime')
         query_type = self.kwargs['type']
         query_type = query_type.replace('/', '')
         if query_type == 'current':
-            return Event.objects.filter(startTime__lte=current_time, endTime__gte=current_time,
-                                        status=Event.STATUS_ACTIVE).order_by('postingTime')
+            logger.debug('Type is current')
+            query_set = Event.objects.filter(startTime__lte=current_time, endTime__gte=current_time,
+                                             status=Event.STATUS_ACTIVE).order_by('postingTime')
         if query_type == 'upcoming':
+            logger.debug('Type is upcoming')
             one_week_out = current_time + datetime.timedelta(days=7)
-            return Event.objects.filter(startTime__range=[current_time, one_week_out],
-                                        status=Event.STATUS_ACTIVE).order_by('postingTime')
-        else:
-            return Event.objects.all().order_by('postingTime')
+            query_set = Event.objects.filter(startTime__range=[current_time, one_week_out],
+                                             status=Event.STATUS_ACTIVE).order_by('postingTime')
+        # else:
+        #     logger.debug('Type is ELSE')
+        #     query_set = Event.objects.all().order_by('postingTime')
+        for event in query_set:
+            if event.owner == me:
+                event.owner = True
+            else:
+                event.owner = False
+
+        return query_set
+
+    def log_action(self):
+        me = self.request.user.__str__()
+        method = self.request.META['REQUEST_METHOD']
+        url = self.request.META['PATH_INFO']
+        # EXAMPLE - dovek : GET /api/events/current/
+        message = '%s : %s %s' % (me, method, url)
+        logger.info(message)
+        return me
```

Comment:
I wouldn't expect a method called `log_action` to return anything, especially not something like a user name.